src/script.c

```diff
@@ -262,7 +262,7 @@ static int scriptVerifyWriteCommandAllow(scriptRunCtx *run_ctx, char **err) {
      * of this script. */
     int deny_write_type = writeCommandsDeniedByDiskError();
 
-    if (server.masterhost && server.repl_slave_ro && run_ctx->original_client->flags != CLIENT_ID_AOF
+    if (server.masterhost && server.repl_slave_ro && run_ctx->original_client->id != CLIENT_ID_AOF
```

Comment:
:scream: i wonder how why the tests didn't catch it when we broke it.
do you happen to know?

Reply:
No, maybe it just wasn't covered by the test `EVAL processes writes from AOF in read-only slaves` (It should be). I think maybe this cases was missed when effects replication was introduced?

Reply:
looks like it was covered by that test, and and indeed it disabled preamble-rdb, and, used DEBUG LOADAOF that empties the old db, **but it looks like the test forgot to disable `lua-replicate-commands`**, so since that feature was made the default, this test was ineffective.

luckily, now that we decided to completely drop support for disabling lua-replicate-commands, we realized that this test can no longer work by letting the EVAL commands generate the AOF, and we moved to generate it on our own, and found that bug.

and we're also lucky that the bug that checked the wrong variable is not yet released.